src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -1326,15 +1326,43 @@ impl_commit_tree (RpmOstreeTreeComposeContext *self,
   if (self->treefile)
     {
       g_autoptr(GFile) treefile_dirpath = g_file_get_parent (self->treefile_path);
+      g_autoptr(GPtrArray) sysuser_entries = NULL;
       if (!rpmostree_check_passwd (self->repo, self->rootfs_dfd, treefile_dirpath, self->treefile,
-                                   self->previous_checksum,
+                                   self->previous_checksum, &sysuser_entries,
```

Comment:
It seems to me that if we're using sysusers - there's really no reason to go into any of the `rpmostree_check_passwd()` code.  With sysusers, the uids are assigned on the client side, unlike every other approach here.  The code here was mostly concerned with comparing from the "previous" values, but conceptually we don't have any previous values.

It seems the main thing we're lifting there is the current values of the passwd/group database.  

One thing it's tempting to do here too is add a new `systemd-sysusers: true` boolean, and then that overrides all of the other legacy stuff.  Then here in `compose-builtin-tree.c` we'd do:
```
if (self->treefile)
  {
    if (sysusers)
      {
        if (!rpmostree_passwd_sysusers_convert (self->rootfs_fd, cancellable, error))
          return FALSE;
      }
    else
      {
        if (!rpmostree_check_passwd (self->repo, self->rootfs_dfd, ...)
          return FALSE;
      }
   }
```

or so?

Reply:
>It seems to me that if we're using sysusers - there's really no reason to go into any of the rpmostree_check_passwd() code. With sysusers, the uids are assigned on the client side, unlike every other approach here.

Ha, I was thinking to propose this approach, but then I figured some packages seem to own their specific uid/gid (Can't seem to find a source, so may be I was wrong).  One example i believe is `daemon` sometimes own uid 2?  So, I thought `check_passwd` and `check_group` was mainly used for that case (the exact mapping of package assignment). 

But if not, then I agree, we don't necessarily need to check sysusers for "previous" values.

Reply:
> One thing it's tempting to do here too is add a new systemd-sysusers: true boolean, and then that overrides all of the other legacy stuff. Then here in compose-builtin-tree.c we'd do:

Yea, this implementation makes sense, and if we are skipping that, may be we can also skip the sections here too? 
https://github.com/projectatomic/rpm-ostree/blob/master/src/libpriv/rpmostree-postprocess.c#L971-L991.

I was thinking earlier if we could work around the sysuser  logic a bit so that the migration of passwd needs not to happen (or skipped) in postprocess.

Reply:
> Ha, I was thinking to propose this approach, but then I figured some packages seem to own their specific uid/gid (Can't seem to find a source, so may be I was wrong). One example i believe is daemon sometimes own uid 2?

A bunch of uid/gid values are fixed; see https://fedoraproject.org/wiki/Packaging:UsersAndGroups and note https://pagure.io/setup/blob/master/f/uidgid

Is that what you're thinking of?

But that...oh wait, argh, I think this calls into question our entire approach here :frowning:  - I'll comment below.

Reply:
>A bunch of uid/gid values are fixed; see https://fedoraproject.org/wiki/Packaging:UsersAndGroups and note https://pagure.io/setup/blob/master/f/uidgid
Is that what you're thinking of?

Yup. :D